ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -783,18 +779,21 @@ int mca_pml_ucx_probe(int src, int tag, struct ompi_communicator_t* comm,
     ucp_tag_t ucp_tag, ucp_tag_mask;
     ucp_tag_recv_info_t info;
     ucp_tag_message_h ucp_msg;
+    int i;
 
     PML_UCX_TRACE_PROBE("probe", src, tag, comm);
 
     PML_UCX_MAKE_RECV_TAG(ucp_tag, ucp_tag_mask, tag, src, comm);
-    for (;;) {
-        ucp_msg = ucp_tag_probe_nb(ompi_pml_ucx.ucp_worker, ucp_tag, ucp_tag_mask,
-                                   0, &info);
-        if (ucp_msg != NULL) {
-            mca_pml_ucx_set_recv_status_safe(mpi_status, UCS_OK, &info);
-            return OMPI_SUCCESS;
+    while (1) {
```

Comment:
let's also update iprobe/improbe/mprobe
the non-blocking versions should trigger opal_progress 1/progress_iterations of the times [if not found a match]

Reply:
didn't get - add loop for UCX/opal progress calling into non-blocking calls?

Reply:
today non-blocking probe calls opal_progress
we want it to call (g_count % progress_iterations) ? ucx_progress : opal_progress

Reply:
it will require additional static counter inside of routine, is it ok?

Reply:
yes, the counter does not even have to be thread-safe for correctness

Reply:
done